File: cpp/alerts/peer_connect.cpp
#include "./peer_connect.h"

namespace ltrs {

uint8_t peer_connect_alert_get_direction(lt::peer_connect_alert* alert) {
    return static_cast<uint8_t>(alert->direction);
}

uint8_t peer_connect_alert_get_socket_type(lt::peer_connect_alert* alert) {
    return static_cast<uint8_t>(alert->socket_type);
}

} // namespace ltrs

File: src/ffi/alerts/peer_connect.rs
#[cxx::bridge(namespace = "ltrs")]
pub(crate) mod ffi {
    #[namespace = "libtorrent"]
    extern "C++" {
        type peer_connect_alert = crate::ffi::ffi::peer_connect_alert;
    }

    unsafe extern "C++" {
        include!("cpp/alerts/peer_connect.h");
        unsafe fn peer_connect_alert_get_direction(alert: *mut peer_connect_alert) -> u8;
        unsafe fn peer_connect_alert_get_socket_type(alert: *mut peer_connect_alert) -> u8;
    }
}
